toolchain/check/scope_stack.h

```diff
@@ -210,6 +224,8 @@ struct ScopeStack::SuspendedScope {
   // to keep the size of a `SuspendedFunction` reasonable while avoiding heap
   // allocations most of the time.
   llvm::SmallVector<LexicalLookup::SuspendedResult, 8> suspended_lookups;
+  // The compile-time bindings that were introduced in this scope.
+  llvm::SmallVector<SemIR::InstId, 8> compile_time_bindings;
```

Comment:
Why 8? There's a comment on suspended_lookups, but it doesn't clearly apply to this member.

If this is because we expect it to be 1:1 (or thereabout) with LexicalLookup::SuspendedResult, could they be combined? In part, I think there's a benefit to avoiding an additional vector if they're expected to be similar size; for things with a few parameters, even with no compile time bindings, it'll be smaller or equivalent just because of vector overhead.

Reply:
Hmm, we could almost avoid storing the compile time bindings entirely, but the list of bindings and the list of lookups can be out of sync due to error recovery. For
```
fn F(T:! type, T:! type)
```
we have two compile-time bindings but only one lookup result. I've combined these into a single vector as suggested.